lib/money/money.rb

```diff
@@ -189,6 +189,52 @@ def allocate(splits)
     amounts.collect { |cents| Money.from_cents(cents) }
   end
 
+  # Allocates money between different parties up to the maximum amounts specified.
+  # Left over pennies will be assigned round-robin up to the maximum specified.
+  # Pennies are dropped when the maxima are attained.
+  #
+  # @example
+  #   Money.new(30.75).allocate_amounts([Money.new(26), Money.new(4.75)])
+  #     #=> [Money.new(26), Money.new(4.75)]
+  #
+  #   Money.new(30.75).allocate_amounts([Money.new(26), Money.new(4.74)]
+  #     #=> [Money.new(26), Money.new(4.74)]
+  #
+  #   Money.new(30).allocate_amounts([Money.new(15), Money.new(15)]
+  #     #=> [Money.new(15), Money.new(15)]
+  #
+  #   Money.new(1).allocate_amounts([Money.new(33), Money.new(33), Money.new(33)])
+  #     #=> [Money.new(0.34), Money.new(0.33), Money.new(0.33)]
+  #
+  #   Money.new(100).allocate_amounts([Money.new(5), Money.new(2)])
+  #     #=> [Money.new(5), Money.new(2)]
+  def allocate_amounts(maxima)
```

Comment:
- I know the latin plural of maximum is maxima, but maximums is accepted in english
- Should this just be called `allocate_discounts` or `allocate_to_max` or something? 
  - It then follows that the argument could be called `amounts`